File: app/src/main/java/com/benhero/glstudio/base/GLAlphaAnimation.java
package com.benhero.glstudio.base;

/**
 * GL透明度动画
 *
 * @author chenbenbin
 */
public class GLAlphaAnimation extends GLAnimation {
    private float mFromAlpha;
    private float mToAlpha;

    public float getFromAlpha() {
        return mFromAlpha;
    }

    public void setFromAlpha(float fromAlpha) {
        mFromAlpha = fromAlpha;
    }

    public float getToAlpha() {
        return mToAlpha;
    }

    public void setToAlpha(float toAlpha) {
        mToAlpha = toAlpha;
    }

    @Override
    protected void applyTransformation(float percent, GLObject object) {
        super.applyTransformation(percent, object);
        object.setAlpha(percent);
    }
}

File: app/src/main/java/com/benhero/glstudio/base/GLAlphaAnimation.kt
package com.benhero.glstudio.base

/**
 * GL透明度动画
 *
 * @author chenbenbin
 */
class GLAlphaAnimation : GLAnimation() {
    var fromAlpha: Float = 0.toFloat()
    var toAlpha: Float = 0.toFloat()

    override fun applyTransformation(percent: Float, `object`: GLObject) {
        super.applyTransformation(percent, `object`)
        `object`.alpha = fromAlpha + (toAlpha - fromAlpha) * percent
    }
}
